bin/all_sky_search/pycbc_coinc_hdfinjfind

```diff
@@ -194,31 +206,43 @@ for trigger_file, injection_file in zip(args.trigger_files,
 
     found_fore = numpy.arange(0, len(stat), 1)[left[found]]
     found_fore_v = numpy.arange(0, len(stat), 1)[left[found_after_vetoes]]
+
     logging.info('Saving injection information')
-    columns = ['mass1', 'mass2', 'spin1x', 'spin1y',
-               'spin1z', 'spin2x', 'spin2y', 'spin2z',
-               'eff_dist_l', 'eff_dist_h', 'eff_dist_v',
-               'inclination', 'polarization', 'coa_phase',
-               'latitude', 'longitude', 'distance']
-    xml_to_hdf(sim_table, fo, 'injections', columns)
-    hdf_append(fo, 'injections/end_time', inj_time)
-
-    # pick up optimal SNRs
-    for ifo, column in args.optimal_snr_column.items():
-        optimal_snr_all = numpy.array(sim_table.get_column(column))
-        # As a single detector being vetoed won't veto all combinations,
-        # need to set optimal_snr of a vetoed ifo to zero in order
-        # to later calculate decisive optimal snr
-        iws, _ = indices_within_segments(inj_time, [args.veto_file], ifo=ifo,
-                                         segment_name=args.segment_name)
-        optimal_snr_all[iws] = 0
-        hdf_append(fo, 'injections/optimal_snr_%s' % ifo,
-                   optimal_snr_all)
-
-    # pick up redshift
-    if args.redshift_column:
-        hdf_append(fo, 'injections/redshift',
-                   sim_table.get_column(args.redshift_column))
+    if '.xml' or '.xml.gz' in injection_file:
+        ninj = len(sim_table)
+        columns = ['mass1', 'mass2', 'spin1x', 'spin1y',
+                   'spin1z', 'spin2x', 'spin2y', 'spin2z',
+                   'inclination', 'polarization', 'coa_phase',
+                   'latitude:dec', 'longitude:ra', 'distance']
+        xml_to_hdf(sim_table, fo, 'injections', columns)
+        hdf_append(fo, 'injections/tc', inj_time)
+
+        # pick up optimal SNRs
+        for ifo, column in args.optimal_snr_column.items():
+            # As a single detector being vetoed won't veto all combinations,
+            # need to set optimal_snr of a vetoed ifo to zero in order
+            # to later calculate decisive optimal snr
+            optimal_snr_all = numpy.array(sim_table.get_column(column))
+            optimal_snr_all[veto_dict[ifo]] = 0
+            hdf_append(fo, 'injections/optimal_snr_%s' % ifo,
+                       optimal_snr_all)
```

Comment:
This adds changes to the XML code pth, but avoids a potentially expensive repetition, so good!